### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Bestbynature**
## Features included in this pull request are as follows:

- [x] A new column `species` of type `string` was added to the animals' table and the schema.sql file was updated.
- [x] The given data was inserted into the table and the data.sql file was updated
- [x] Inside a transaction, the animals' table was updated by setting the species column to unspecified and verified before rolling it back.
- [x] Inside a transaction:

- The animals' table was updated by setting the species column to digimon for all animals that have a name ending with 'mon'.
- The animals' table was updated by setting the species column to pokemon for all animals that don't have species already set and the changes were committed.

digimon-pokemon

- [x] Inside a transaction,  all records in the animals' table were deleted and then the transaction was rolled back.

delete-records

- [x] Inside a transaction:

- All the records of animals born after Jan 1st, 2022 were deleted.
- A savepoint was created for the transaction.
- All animals' weights were updated to be their weight multiplied by -1.
- A Rollback was made to the created savepoint
- All animals' weights that are negative were updated to be their weight multiplied by -1 and the transaction was committed and added to the `queries.sql` file.

savepoint-delete-after-jan

- [x] Queries were written to answer the following questions:

- How many animals are there?
count-function

- How many animals have never tried to escape?
esc-att-equals-zero

- What is the average weight of animals?
avg-weight

- Who escapes the most, neutered or not neutered animals?
esc-count-neutered-or-not

- What is the minimum and maximum weight of each type of animal?
min-max-weight-species-group

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
avg-esc-1990-dob

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -5,3 +5,5 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (2, 'Gabumon', '2018-11-15', 2, '1', 8.00), 
 (3, 'Pikachu', '2021-01-07', 1, '0', 15.04), 
 (4, 'Devimon', '2017-05-12', 5, '1', 11.00);
+
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (5, 'Charmander', '2020-02-08', 0, '0', 11.00), (6, 'Plantmon', '2021-11-15', 2, '1', 5.70), (7, 'Squirtle', '1993-04-02', 3, '0', 12.13), (8, 'Angemon', '2005-06-12', 1, '1', 45.00), (9, 'Boarmon', '2005-06-07', 7, '1', 20.40), (10, 'Blossom', '1998-10-13', 3, '1', 17.00), (11, 'Ditto', '2022-05-14', 4, '1', 22.00);
```
Comment (ShadyShawkat) [inline data.sql]: - [x] Kindly make sure to insert the right data. Kindly change the first four `weight_kg` values to be negatives to match the given data.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Bestbynature, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (Bestbynature) [reply data.sql]: Thank you so much @ShadyShawkat. The correction has been effected as advised.

**Review by Munsa1 — APPROVED** [review]
Hi @Bestbynature, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 


## Optional suggestions
### Important :red_circle: 
- [x] It would be syntactically correct if you gave the `neutered` field a DATATYPE of `boolean` so that if an animal is neutered the result returned would be `true` otherwise it would be `false`. This is less confusing for the user in my opinon.
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**